pkg/cmd/secrets.go

```diff
@@ -380,6 +408,85 @@ func downloadSecrets(cmd *cobra.Command, args []string) {
 	utils.Log(fmt.Sprintf("Downloaded secrets to %s", filePath))
 }
 
+func templateSecrets(cmd *cobra.Command, args []string) {
+	localConfig := configuration.LocalConfig(cmd)
+
+	utils.RequireValue("token", localConfig.Token.Value)
+
+	filePath, err := utils.GetFilePath(args[0])
+	if err != nil {
+		utils.HandleError(err, "Unable to parse template file path")
+	}
+
+	var file []byte
+	file, err = ioutil.ReadFile(filePath)
+	if err != nil {
+		utils.HandleError(err, "Unable to read template file")
+	}
+
+	var outputFilePath string
+	output := cmd.Flag("output").Value.String()
+	if len(output) != 0 {
+		outputFilePath, err = utils.GetFilePath(output)
+		if err != nil {
+			utils.HandleError(err, "Unable to parse output file path")
+		}
+	}
+
+	funcs := map[string]interface{}{
+		"tojson": func(value string) (string, error) {
+			body, err := json.Marshal(value)
+			if err != nil {
+				return "", err
+			}
+			return string(body), nil
+		},
+		"fromjson": func(value string) (interface{}, error) {
+			var result interface{}
+			err = json.Unmarshal([]byte(value), &result)
+			if err != nil {
+				return "", err
+			}
+			return result, nil
+		},
+	}
+	template, err := template.New("Secrets").Funcs(funcs).Parse(string(file))
+	if err != nil {
+		utils.HandleError(err, "Unable to parse template text")
+	}
+
+	response, responseErr := http.GetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+	if !responseErr.IsNil() {
+		utils.HandleError(responseErr.Unwrap(), responseErr.Message)
+	}
+
+	secrets, parseErr := models.ParseSecrets(response)
+	if parseErr != nil {
+		utils.HandleError(parseErr, "Unable to parse API response")
+	}
+
+	secretsMap := map[string]interface{}{}
+	for _, secret := range secrets {
+		secretsMap[secret.Name] = secret.ComputedValue
+	}
+
+	buffer := new(strings.Builder)
+	err = template.Execute(buffer, secretsMap)
+	if err != nil {
+		utils.HandleError(err, "Unable to render template")
+	}
+
+	if outputFilePath != "" {
+		err = ioutil.WriteFile(outputFilePath, []byte(buffer.String()), 0644)
```

Comment:
```suggestion
		err = ioutil.WriteFile(outputFilePath, []byte(buffer.String()), 0600)
```

This would allow all users to read these secrets 😢

Reply:
Oh dear -- Egg on my face, sorry about that

Reply:
All good, that's why we have gosec :)